lib/assert-result-factory.js

```diff
@@ -0,0 +1,33 @@
+/** @babel */
+
+class AssertResultFactory {
```

Comment:
As always @sindresorhus, thank you so much for your feedback on this. Not sure if you're going to have time for this, but here goes my question:

Wouldn't that put too much logic in TestRunnerProcess (the client of the factory)?. Also, not 100% sure, but I'd say slightly harder to test, bigger setup needed and probably harder to understand.

Anyway, I'm already implemented the change, but would love to hear your thoughts.